contracts/consensus/Consensus.sol

```diff
@@ -317,144 +301,232 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
+        if (coreLifetimes[address(core)] == CoreLifetime.genesis) {
+            coreLifetimes[address(core)] = CoreLifetime.active;
+        }
     }
 
     /**
-     * @notice Forms a new committee to verify the precommit proposal.
+     * @notice Forms a new committee to verify the precommit.
      *
      * @dev Function requires:
-     *          - core has precommitted
-     *          - the current block height is bigger than the precommitt's
-     *            committee formation height
+     *          - the given metachain id is not 0
+     *          - assigned core has precommitted a metablock
+     *          - committee formation height passed
      *          - committee formation blocksegment must be in the most
      *            recent 256 blocks.
-     *
-     * @param _core Core contract address.
      */
-    function formCommittee(address _core)
+    function formCommittee(bytes32 _metachainId)
         external
     {
         require(
-            coreLifetimes[_core] == CoreLifetime.active,
-            "Core lifetime status must be active"
+            _metachainId != bytes32(0),
+            "Metachain id is 0."
         );
 
-        Precommit storage precommit = precommits[_core];
+        uint256 currentHeight = metablockTips[_metachainId];
+        Metablock storage currentMetablock = metablockchains[_metachainId][currentHeight];
+
         require(
-            precommit.proposal != bytes32(0),
-            "Core has not precommitted."
+            currentMetablock.round == MetablockRound.Precommitted,
+            "Assigned core must have precommitted a metablock to form a committee."
         );
 
-        require(
-            block.number > precommit.committeeFormationBlockHeight,
-            "Block height must be higher than set committee formation height."
+        uint256 committeeFormationBlockHeight = currentMetablock.roundBlockNumber.add(
+            COMMITTEE_FORMATION_DELAY
         );
 
         require(
-            block.number <= precommit.committeeFormationBlockHeight
-                .sub(COMMITTEE_FORMATION_LENGTH)
-                .add(uint256(256)),
-            "Committee formation blocksegment is not in most recent 256 blocks."
+            block.number > committeeFormationBlockHeight,
+            "Committee formation height has not yet come to pass."
         );
 
-        uint256 segmentHeight = precommit.committeeFormationBlockHeight;
-        bytes32[] memory seedGenerator = new bytes32[](uint256(COMMITTEE_FORMATION_LENGTH));
-        for (uint256 i = 0; i < COMMITTEE_FORMATION_LENGTH; i = i.add(1)) {
-            seedGenerator[i] = blockhash(segmentHeight);
-            segmentHeight = segmentHeight.sub(1);
-        }
-
-        bytes32 seed = keccak256(
-            abi.encodePacked(seedGenerator)
+        bytes32 seed = hashBlockSegment(
+            committeeFormationBlockHeight,
+            COMMITTEE_FORMATION_LENGTH
```

Comment:
why are we passing the constant to an internal function; in 

consider instead passing the `metablockHash`, because otherwise if multiple precommits happen in the same Ethereum block, they will all have the same seed, and thus the same committee. 

cc @deepesh-kn @jayeshbairagi can we make this a separate task?
```
        bytes32 seed = hashBlockSegments(
            currentMetablock.metablockHash,
            committeeFormationHeight);
```

Reply:
Done.